### DISCUSSION TITLE
Integration of corcua dataset library

### DISCUSSION THREAD

**Discussion by DanBmh** [speechbrain/speechbrain#1626] [Q&A]
Hi,

I wanted to notify you about my speech dataset handling tool [corcua](https://gitlab.com/Jaco-Assistant/corcua). The task it solves is loading datasets from different formats and converting them to a format we can use for training STT networks. It was built with the goal of making the addition of new datasets as easy as possible. Currently it does support 34 datasets, many in  German, but 12 of them are multilingual containing 3 or more languages.

There would be two possible options for integration:
1. allow _corcua's_ default format as input format. This is 16kHz 16-bit wav files and a `tab` separated _csv_ file with at least the keys `text`, `duration`, `filepath` and arbitrary optional keys (depending on the dataset). 
2. adding a new _writer_ for Speechbrain's custom format, I would estimate this will take about 50 lines of code.

I hope this is interesting for your project.
Greetings
Daniel

**Comment by TParcollet** [comment]
Hi @DanBmh, this is a great initiative. Maybe option 2 could be something feasible. For a good integration, we would require you to have something that creates the json / csv files that SpeechBrain requires. This would be the simplest way of achieving this.

**Comment by DanBmh** [comment]
Hi @TParcollet , thanks for your answer.
I also think adding a new _writer_ is very easy, but I'm not sure about the format SpeechBrain requires. Maybe you or someone else could write the _writer_ class? An example for DeepSpeech's format is [here](https://gitlab.com/Jaco-Assistant/corcua/-/blob/master/corcua/writers/deepspeech.py). 

If there are no special requirements regarding the audio file, the standard workflow (for the DeepSpeech example too) is loading the dataset, saving it in _corcua's_ default format, and then converting the _csv_ files to your custom format. Else the audio handling would need to be implemented too, similar to the [basewriter](https://gitlab.com/Jaco-Assistant/corcua/-/blob/master/corcua/writers/base_writer.py) class.

The internal dataset representation of corcua is a list of dictionaries:
```
[
  {
     "filepath": "/path/to/audio.xyz",
     "text": "the transcription",
     "duration": 1.23,
     "arbitrary key": "some data",
     ...
  },
  ...
]
```

**Comment by Adel-Moumen** [comment]
Hello @DanBmh,

Sorry for the very late response. 

I saw on your gitlab project that you wrote the speechbrain writer class? https://gitlab.com/Jaco-Assistant/corcua/-/blob/master/corcua/writers/speechbrain.py 

Therefore, it seems that the issue is no longer an issue. Could you please tell me if you have any other questions or can we close this issue? Many thanks, and congratulations on this great initiative. 😃 

Best.

**Comment by DanBmh** [comment]
You're right, I didn't had the time to notify you the last days...
It also seems that this helps with/solves your issues #1210 and #1198.

It would be great if we could add a note somewhere in Speechbrain's readme.

**Comment by Adel-Moumen** [comment]
Hello @TParcollet,

Could you please take a look at this? Thanks. 🙂
